### PR TITLE
Logging UI - Introspection, Regex Search, WHERE Querying

### PR DESCRIPTION
**Pull request by Ziinc**
### Logging UI
This PR introduces the following additional features to Supabase Platform

Demo:

https://user-images.githubusercontent.com/22714384/144467543-74fa8e19-0030-4c91-8f7c-0e1eeafc1c01.mov

#### Log introspection
Upon selecting a log line, the log selection pane will open, displaying log details and metadata.

Logging display is enabled though the forked react-data-grid package, allowing a similar ux to sql results display.

Keyboard shortcuts are enabled, allowing quick browsing through each log line.

#### Regex searching
Regex searching is possible through the search bar, allowing easy log filtering. This allows for simple case sensitive searching

#### Query Presets
To allow simple and accessible log querying, regex query preset buttons have been provided, that automatically populate the search bar.

#### Background New Logs Polling
To allow users to monitor new incoming logs, a `Load new logs` button appears when new logs are detected.

#### WHERE clause custom querying
`WHERE` clause querying is possible through the `Custom Query` button. It will open a stripped-down monaco editor with pgsql syntax highlighting that will allow users to input custom filters, allowing timestamp and metadata filtering etc.

### CODE REVIEW

**Comment by luca400** [comment]
Not an expert, or even a contributor, but nevertheless I checked out your pr and lgtm

**Comment by Ziinc** [comment]
PR ready for review, tests have been updated with latest ux changes. Needs to be coordinated with platform PR merge.

**Comment by Ziinc** [comment]
Resolved merge conflicts and rebased on master.

File: studio/pages/project/[ref]/settings/logs/[type].tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+            language="pgsql"
+            defaultValue={where}
+            onInputChange={(v) => setWhere(v || '')}
+            onInputRun={handleRefresh}
+          />
+        </div>}
+      {error && (
+        <Typography.Text className="text-center w-full block">Could not fetch data</Typography.Text>
+      )}
+      {newCount ? <Button type="dashed" onClick={handleRefresh} block>Load new logs</Button> : null}
```
Comment (phamhieu) [inline studio/pages/project/[ref]/settings/logs/[type].tsx:103]: extract this button to a component. cleaner code.

File: studio/tests/pages/projects/LogTable.test.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+      my_key: "something_value"
+    }
+  }]} />)
+
+  await waitFor(() => screen.getByText(/happened/))
+  const row = screen.getByText(/happened/)
+  fireEvent.click(row)
+  await waitFor(() => screen.getByText(/my_key/))
+  await waitFor(() => screen.getByText(/something_value/))
+})
```
Comment (phamhieu) [inline studio/tests/pages/projects/LogTable.test.js:18]: Nice!!!

**Review by phamhieu — APPROVED** [review]
lgtm!

File: studio/pages/project/[ref]/settings/logs/[type].tsx (hunk 1 above)
Comment (Ziinc) [reply studio/pages/project/[ref]/settings/logs/[type].tsx:113]: Noted, will defer the swr data refactoring to when i'll be adding in the 'Load older logs' button, as logic will have to be adjusted anyway.

**Pull request merged**